components/autocomplete/index.js

```diff
@@ -3,6 +3,7 @@
  */
 import classnames from 'classnames';
 import { escapeRegExp, find, filter, map } from 'lodash';
+import 'element-closest';
```

Comment:
`wp.components.Autocomplete` was exposed for some time. Is there any way to make it backward compatible for the next 2 release cycles? We should also add a deprecation message to give a warning about the changes.

Reply:
We should be able to detect a difference in the completer objects and log a deprecation message then. Do we want to limit such messages or print for every completer object seen with the old interface?

Reply:
In other cases, it was done per function/component occurrence. See #5833 for reference.